kubernetes/samples/scripts/domain-lifecycle/waitForDomain.sh

```diff
@@ -0,0 +1,598 @@
+  #   - silently set global to "" if EXPECTED_STATE = '0'
+  #   - "exit 1" and report an error if EXPECTED_STATE = 'Completed'
+
+  getDomainValue    domain_info_goal_RV_current        ".spec.restartVersion"
+  getDomainValue    domain_info_goal_IV_current        ".spec.introspectVersion"
+  getDomainValue    domain_info_goal_image_current     ".spec.image"
+  getDomainValue    domain_info_clusters               ".spec.clusters"
+  getDomainAIImages domain_info_goal_aiimages_current
+  getDomainValue    domain_info_api_version            ".apiVersion"
+  getDomainValue    domain_info_condition_failed_str   ".status.conditions[?(@.type==\"Failed\")]" # has full failure messages, if any
+  getDomainValue    domain_info_condition_completed    ".status.conditions[?(@.type==\"Completed\")].status" # "True" when complete
+  getDomainValue    domain_info_observed_generation    ".status.observedGeneration"
+
+  domain_info_clusters=$( echo "$domain_info_clusters" | sed 's/"name"//g' | tr -d '[]{}:' | sortlist | sed 's/,/ /') # convert to sorted space separated list
+
+  # gather observed and goal generation for each cluster:
+  cl_info_observed_generations=""
+  cl_info_goal_generations=""
+  local lgen
+  local cl
+  for cl in $domain_info_clusters EOL; do
+    [ "$cl" = "EOL" ] && break
+    getClusterValue $cl lgen ".status.observedGeneration"
+    [ -z "$lgen" ] && lgen="NOTSET"
+    cl_info_observed_generations="$cl_info_observed_generations $lgen"
+
+    getClusterValue $cl lgen ".metadata.generation"
+    cl_info_goal_generations="$cl_info_goal_generations $lgen"
+  done
+
+  # Get domain goal generation last in case gen changed while checking the above...
+  getDomainValue    domain_info_goal_generation        ".metadata.generation"
+
+  all_goals_current="${domain_info_goal_RV_current}
+${domain_info_goal_IV_current}
+${domain_info_goal_image_current}
+${domain_info_goal_aiimages_current}
+${domain_info_goal_generation}
+${domain_info_observed_generation}
+${cl_info_goal_generations}
+${cl_info_observed_generations}
+${domain_info_condition_failed_str}
+${domain_info_clusters}
+^M"
+}
+
+getPodInfo() {
+  # get interesting info about each pod in DOMAIN_NAMESPACE with DOMAIN_UID
+  # output: stdout in the form "var=;value; var2=;value; ..." (one line per pod)
+  #
+  # NOTE: ljpath must correspond with the regex below and column headers below 
+
+  local ljpath=''
+  ljpath+='{range .items[*]}'
+    ljpath+='{" name="}'
+    ljpath+='{";"}{.metadata.name}{";"}'
+    ljpath+='{" domainRestartVersion="}'
+    ljpath+='{";"}{.metadata.labels.weblogic\.domainRestartVersion}{";"}'
+    ljpath+='{" introspectVersion="}'
+    ljpath+='{";"}{.metadata.labels.weblogic\.introspectVersion}{";"}'
+    ljpath+='{" image="}'
+    ljpath+='{";"}{.spec.containers[?(@.name=="weblogic-server")].image}{";"}'
+    ljpath+='{" aiimages="}'
+    ljpath+='{";"}{.spec.initContainers[?(@.command[0]=="/weblogic-operator/scripts/auxImage.sh")].image}{";"}'
```

Comment:
Will this work if there are multiple AI images? Do we need a comma after the image name?

Reply:
> Will this work if there are multiple AI images?

Yes. It will return a space separated list.

> Do we need a comma after the image name?

No. The subsequent pipe to 'sortAIImages' handles the space and, once sorted, replaces the space with commas.

Reply:
ok, thanks.